lib/libc/Kconfig

```diff
@@ -79,6 +72,9 @@ config MINIMAL_LIBC
 	imply COMPILER_FREESTANDING
 	select COMMON_LIBC_ABORT
 	select COMMON_LIBC_STRNLEN
+	imply COMMON_LIBC_MALLOC
```

Comment:
Until #65434 is merged, I don't think we should be adding malloc to the default minimal libc configuration as that could increase the size of applications which don't use it. That's because the malloc heap initializer will always end up linked into applications because the common C library bits are included under the `--whole-archive` list.

Reply:
I had a change to remove the sys_init from the minimal libc malloc (common, I guess) and use lazy init instead (init on first use), but it was more complicated than that.

Honestly, that should have a bug / issue number attached to it, because it needs to be fixed. In no universe should we be unconditionally linking malloc buffers and code without an explicit call to malloc or something related.

Reply:
This is moved from:

```
...

if MINIMAL_LIBC

...

config MINIMAL_LIBC_MALLOC
	bool "[DEPRECATED] Minimal libc malloc implementation"
	default y
	imply COMMON_LIBC_MALLOC
	help
	  [DEPRECATED] Use COMMON_LIBC_MALLOC

	  Enable the minimal libc's implementation of malloc, free, and realloc.
	  Disable if you wish to provide your own implementations of these functions.

...

```

which I have deleted in `lib/libc/minimal/Kconfig`

Reply:
Yup, but this changes the default. Before, you didn't get malloc unless you asked for it. With the proposed change, you get malloc unless you ask to *not* have it. Unless I'm mis-reading the patch?

Reply:
Currently, enabling `MINIMAL_LIBC` would enable `MINIMAL_LIBC_MALLOC` by default, which would `imply` `COMMON_LIBC_MALLOC`, so malloc will be enabled unless user explicitly disables `MINIMAL_LIBC_MALLOC` and/or `COMMON_LIBC_MALLOC`.

FWIU, this PR removes the intermediary `MINIMAL_LIBC_MALLOC` & relocate the `imply COMMON_LIBC_MALLOC` directly under the `MINIMAL_LIBC`, so `COMMON_LIBC_MALLOC` will be enabled by default, which is the same as before, unless explicitly disabled by the user, but there's one less Kconfig (`MINIMAL_LIBC_MALLOC`) that a user has to care.

Or maybe I missed something?

Reply:
Nope, *I* missed that minimal libc sets `MINIMAL_LIBC_MALLOC` to `default y`. Thanks for your patience, this looks fine.